kratos/geometries/nurbs_curve_on_surface_geometry.h

```diff
@@ -155,6 +160,56 @@ class NurbsCurveOnSurfaceGeometry : public Geometry<typename TSurfaceContainerPo
             << std::endl;
     }
 
+    ///@}
+    ///@name Access to Geometry Parts
+    ///@{
+
+    /**
+    * @brief This function returns the pointer of the geometry
+    *        which is corresponding to the trim index.
+    *        Possible indices are:
+    *        SURFACE_INDEX or EMBEDDED_CURVE_INDEX.
+    * @param Index: SURFACE_INDEX or EMBEDDED_CURVE_INDEX.
+    * @return pointer of geometry, corresponding to the index.
+    */
+    GeometryPointer pGetGeometryPart(const IndexType Index) override
+    {
+        const auto& const_this = *this;
+        return std::const_pointer_cast<GeometryType>(
```

Comment:
same

Reply:
Const casting the const. It does avoid coffee duplication.

Reply:
ahh now I understand what this does
you don't want to implement the function twice, once for with and once for without `const`

Usually we duplicate the code but this is already done in other places so fine for me
If we want to change it at some point would be easy

Reply:
Ups too much coffee und too less sleep ;)

Yes, it's not really beautiful. But it does it's job...